linked-lists/c-or-cpp/merge.cpp

```diff
@@ -0,0 +1,97 @@
+#include <iostream>
+
+template <typename T>
+class LinkedList{
+	public:
+		int size = 0;
+		void pushfront(const T& data);
+		T& front();	
+		void popfront();
+		void display();
+		void merge(const LinkedList<T>& list2) ;
+	private:
+		struct node{
+			T data;
+			node* next;
+			node(const T& data): data(data), next(nullptr){}
+		};
+		node* _head = nullptr;
+};
+
+template <typename T>
+T& LinkedList<T>::front(){
+	//returns the first node data of the linked list
+
+	return _head->data;
+}
+
+template <typename T>
+void LinkedList<T>::popfront(){
+	//sets the head pointer to nullptr and the next node as the new head
+	
+	//if there is no node then do nothing
+	if(_head == nullptr) return;
+	if(_head->next == nullptr){
+		_head = nullptr;
+		#delete _head;
+		--size;
+		return;
+	}
+	node* oldhead = _head;
+	_head = _head->next;
+	#delete oldhead;
+	oldhead = nullptr;
+	size--;
+}
+
+template <typename T>
+void LinkedList<T>::pushfront(const T& data){
+	node* n = new node(data);
+	/*if the list is empty pointer the head pointer to 
+	 newly created node */
+	if(_head == nullptr){ _head = n; }
+	else{
+		n->next = _head;
+		_head = n;
+	}
+	size++;
+}
+
+
+template <typename T>
+void LinkedList<T>::display(){
+	node* thru = _head;
+	while(thru != nullptr){
+		std::cout << thru->data << std::endl;
+		thru = thru->next;
+	}
+}
+
+template <typename T>
+void LinkedList<T>::merge(const LinkedList<T>& other) {
+	LinkedList<T> l1 = other;
```

Comment:
As there is no user-defined assignment operator in this class so default operator of the class will operate on here, which will just copy all the contents of `other` to `l1` ( **including the location where head pointer is pointing to** ), this will make the changes that you made to `l1` cause undesired changes in `other` list too.

---

You can fix this problem in 2 ways :- 

1. Create your own custom overloaded assignment operator(`=`) for the class `LinkedList` which does a deep copy of the list.

2. Perform the copy manually here.

Reply:
That bug made me go crazy.
Thanks for the insights.